### PR TITLE
[ci] Run cpp tests in unix_docker_test.sh

### LINKED ISSUE #3682: Move testing C++ from unix_docker_build to unix_docker_test
As @BillXu2000 found out, it's a bit confusing that we currently run C++ tests in the build stage. https://github.com/taichi-dev/taichi/pull/3680/files#diff-038b990ab4476a55a74bb7ed8ed7e10231567d52310b620b5a2884d1faa40dacL48 

And we didn't run C++ tests in tests stage since the taichi_cpp_tests binary was not copied over. I gave a quick try fixing this in #3680 but hit some docker related errors. 

I'll leave it here just in case anyone has bandwidth to improve it, as this issue isn't blocking right now.

### PR DESCRIPTION
**Pull request by knight42**
Related issue = fixes #3682

This PR aims to take over #3680, but I need to do some debugging first.

### CODE REVIEW

**Comment by knight42** [comment]
@ailzhang Hi, looks like the job https://github.com/taichi-dev/taichi/actions/runs/1533675803 is waiting for approval from maintainer, and I need to do some debugging first, could you help me to proceed?

**Comment by knight42** [comment]
@ailzhang Hey! looks like the workflow is running without problems now, should we add the following lines to `unix_docker_test.sh`?
``` bash
CUR_DIR=`pwd`
TI_LIB_DIR=$CUR_DIR/python/taichi/lib ./build/taichi_cpp_tests
```

**Comment by ailzhang** [comment]
@knight42 Awesome thanks! Actually C++ tests already run here ;) 
```
+ python tests/run_tests.py -vr2 -t2 -a cpu,cc
Running main() from /home/dev/taichi/external/googletest/googletest/src/gtest_main.cc
[==========] Running 72 tests from 24 test suites.
[----------] Global test environment set-up.
[----------] 5 tests from AliasAnalysis
[ RUN      ] AliasAnalysis.Alloca
[       OK ] AliasAnalysis.Alloca (1 ms)
[ RUN      ] AliasAnalysis.GlobalPtr_Same
[       OK ] AliasAnalysis.GlobalPtr_Same (1 ms)
```
The trick is https://github.com/taichi-dev/taichi/blob/master/tests/run_tests.py#L216, we turn on C++ test by default and run it if taichi_cpp_tests is present. 

We might need a followup to make sure c++ tests run (right now if you don't copy taichi_cpp_tests over it won't complain :P so we are currently risking not testing C++ by accident.  :P

File: .github/workflows/testing.yml
Comment (ailzhang) [inline .github/workflows/testing.yml:209]: hmmm do we only copy the whls over? We probably also need the build/ dir?

Comment (knight42) [reply .github/workflows/testing.yml:209]: We already mounted the build dir into the container:
```
docker create -v `pwd`/shared/build:/home/dev/build ...
```

**Review by ailzhang — APPROVED** [review]
Thanks for fixing this!

**Comment by knight42** [comment]
@ailzhang Should I squash the commits before merging this?

**Comment by frostming** [comment]
Hi, I am currently working on #3667  and also encountering this problem, it will make some difference but I will refer to your change in this PR.

**Comment by ailzhang** [comment]
@knight42 We used squash and merge by default ;) So it's safe to do so from the webpage :D

**Pull request merged**
